packages/components/select/src/useSelect.ts

```diff
@@ -773,6 +773,100 @@ export const useSelect = (props: SelectProps, emit: SelectEmits) => {
     }
   }
 
+  const isFocusableOption = (arr: any[], index: number) => {
+    const obj = arr[index]
+    return !obj?.isDisabled && obj?.visible
+  }
+
+  const findFocusableIndex = (
+    arr: any[],
+    start: number,
+    step: 1 | -1,
+    len: number
+  ) => {
+    for (let i = start; i >= 0 && i < len; i += step) {
+      if (isFocusableOption(arr, i)) return i
+    }
+    return null
+  }
+
+  const focusOption = (targetIndex: number, mode: 'up' | 'down') => {
+    const len = states.options.size
+    if (len === 0) return
+    const idx = Math.min(Math.max(Math.floor(targetIndex), 0), len - 1)
+    let newIndex = null
+    switch (mode) {
+      case 'up':
+        newIndex =
+          findFocusableIndex(optionsArray.value, idx, -1, len) ??
+          findFocusableIndex(optionsArray.value, idx + 1, 1, len)
+        break
+
+      case 'down':
+        newIndex =
+          findFocusableIndex(optionsArray.value, idx, 1, len) ??
+          findFocusableIndex(optionsArray.value, idx - 1, -1, len)
+        break
+
+      default:
+        newIndex = findFocusableIndex(optionsArray.value, idx, 1, len)
+        break
+    }
```

Comment:
It’s just a suggestion for a cleaner way to write this helper function.

```suggestion
    const start = clamp(targetIndex, 0, len - 1)
    const options = optionsArray.value
    const direction = mode === 'up' ? -1 : 1
    const newIndex =
      findFocusableIndex(options, start, direction, len) ??
      findFocusableIndex(options, start - direction, -direction, len)

```